lib/galaxy/tool_util/cwl/parser.py

```diff
@@ -327,12 +327,8 @@ def _ensure_cwl_job_initialized(self):
                 beta_relaxed_fmt_check=beta_relaxed_fmt_check,
             )
 
-            args = []
+            args = [RuntimeContext(job_args)]
             kwargs: Dict[str, str] = {}
-            if RuntimeContext is not None:
-                args.append(RuntimeContext(job_args))
-            else:
-                kwargs = job_args
```

Comment:
I think the previous code should be restored with a `  # type: ignore[unreachable]` added to this last line, as `RuntimeContext` can be `None` in the tool-util package if the `cwl` extra is not specified.

Reply:
I can do that, but tool parsing would fail in https://github.com/mvdbeek/galaxy/blob/c713b37c588bba50ad7d7897632e36bc16215e4d/lib/galaxy/tool_util/cwl/parser.py#L342 too ... idk if this partial support approach makes sense ?

Reply:
how about

```
diff --git a/lib/galaxy/tool_util/cwl/parser.py b/lib/galaxy/tool_util/cwl/parser.py
index 3207eea965..ac71393c61 100644
--- a/lib/galaxy/tool_util/cwl/parser.py
+++ b/lib/galaxy/tool_util/cwl/parser.py
@@ -293,6 +293,7 @@ class ExpressionToolProxy(CommandLineToolProxy):

 class JobProxy:
     def __init__(self, tool_proxy, input_dict, output_dict, job_directory):
+        assert RuntimeContext is not None, "cwltool is not installed, cannot run CWL jobs"
         self._tool_proxy = tool_proxy
         self._input_dict = input_dict
         self._output_dict = output_dict
```
instead ?

Reply:
Sounds good for me. @jmchilton ?